src/lib/server/orders.ts

```diff
@@ -345,7 +345,10 @@ export async function onOrderPayment(
 export async function onOrderPaymentFailed(
 	order: Order,
 	payment: Order['payments'][0],
-	reason: Exclude<OrderPaymentStatus, Exclude<OrderPaymentStatus, 'canceled' | 'expired'>>
+	reason: Exclude<
+		OrderPaymentStatus,
+		Exclude<OrderPaymentStatus, 'canceled' | 'expired' | 'failed'>
+	>
```

Comment:
Actually you can use `Extract`


```suggestion
	reason: Extract<OrderPaymentStatus, 'canceled' | 'expired' | 'failed'>
```